mapclassify/_classify_API.py

```diff
@@ -214,3 +218,60 @@ def classify(
         classifier = _classifiers[scheme](y, k)
 
     return classifier
+
+
+def classify_to_rgba(
+    values, classifier="quantiles", k=6, cmap="viridis", nan_color=[255, 255, 255, 255]
+):
+    """Convert array of values into RGBA colors using a colormap and classifier.
+
+    Parameters
+    ----------
+    values : list-like
+        array of input values
+    classifier : str, optional
+        string description of a mapclassify classifier, by default "quantiles"
+    k : int, optional
+        number of classes to form, by default 6
+    cmap : str, optional
+        name of matplotlib colormap to use, by default "viridis"
+    nan_color : list, optional
+        RGBA color to fill NaN values, by default [255, 255, 255, 255]
+
+    Returns
+    -------
+    numpy.array
+        array of lists with each list containing four values that define a color using
+        RGBA specification.
+    """
+    if not pd.api.types.is_list_like(nan_color) and not len(nan_color) == 4:
+        raise ValueError("`nan_color` must be list-like of 4 values: (R,G,B,A)")
+
+    # only operate on non-NaN values
+    v = pd.Series(values)
+    legit_indices = v[~v.isna()].index.values
+
+    # transform (non-NaN) values into class bins
+    bins = classify(v.dropna().values, scheme=classifier, k=k).yb
+
+    # create a normalizer using the data's range (not strictly 1-k...)
+    norm = Normalize(min(bins), max(bins))
```

Comment:
I'm not clear why there is a classifier called prior to here. 
It seems to me this is trying to do a classless choropleth map [1].
If that is true, then the first classifier can be omitted and line 258 could become
`norm = Normalize(v.min(), v.max())`

But maybe I'm missing something here in my understanding?


[1] Tobler, W. (1973). Choropleth maps without class intervals. Geographical
Analysis, 5:262-265

Reply:
![image](https://github.com/pysal/mapclassify/assets/118042/7a85f4fe-5fae-4347-b108-74c08350982d)
So do I have this right? 

This was classified k=8, fisher jenks. So I'm thinking there should only be 8 colors in the figure, but the additional z axis allows for differentiation between units in the same bin but with different values. The color of the hex fill would be whatever class the hex value is placed into.

Reply:
Z and color are disconnnected there. Color is based on Jenna but height is continuous

Reply:
(i.e. all this does is take a cmap and translate bins --> colors)

so if you remove the z-dimension, this function gives you the same choropleth geopandas would give you. But then you can also use extrusion to encode the same or a different variable. A bit more at the [end of this](https://knaaptime.com/urban_analysis/02_geoprocessing/01_geovisualization.html#pydeck)